--- brunsli.cmake	2021-10-06 22:13:05.628152000 +0100
+++ brunsli.cmake	2021-10-06 22:13:09.258152000 +0100
@@ -74,7 +74,8 @@
 )
 
 set(BRUNSLI_LIBRARIES brunslicommon-static brunslidec-static brunslienc-static)
-
+set_host_properties(brunslienc-static PROPERTIES OUTPUT_NAME brunslienc-c)
+set_host_properties(brunslidec-static PROPERTIES OUTPUT_NAME brunslidec-c)
 if(NOT BRUNSLI_EMSCRIPTEN)
 add_library(brunslidec-c SHARED
   c/dec/decode.cc
@@ -153,7 +154,7 @@
 # Installation
 if(NOT BRUNSLI_EMSCRIPTEN)
   install(
-    TARGETS brunslidec-c brunslienc-c
+    TARGETS brunslidec-c brunslienc-c brunslidec-static brunslienc-static
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
   )
@@ -171,6 +172,8 @@
   RUNTIME_OUTPUT_DIRECTORY_RELEASE "${CMAKE_BINARY_DIR}/artifacts"
 )
 
+install(TARGETS cbrunsli dbrunsli RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
 if (${BUILD_TESTING})
 
 include(GoogleTest)
