--- src/android/native_app_glue/android_native_app_glue.c	2024-06-09 21:11:16.184348410 +0100
+++ src/android/native_app_glue/android_native_app_glue.c	2024-06-09 21:12:38.548571510 +0100
@@ -17,6 +17,7 @@
 
 #include <jni.h>
 
+#include <stdlib.h>
 #include <errno.h>
 #include <string.h>
 #include <unistd.h>

--- CMakeLists.txt	2024-06-09 21:09:48.359877743 +0100
+++ CMakeLists.txt	2024-06-09 21:10:16.157025282 +0100
@@ -89,7 +89,6 @@
     src/fg_ext.c
     src/fg_font_data.c
     src/fg_gamemode.c
-    src/fg_geometry.c
     src/fg_gl2.c
     src/fg_gl2.h
     src/fg_init.c
@@ -116,6 +115,7 @@
     LIST(APPEND FREEGLUT_SRCS
         src/fg_font.c
         src/fg_menu.c
+        src/fg_geometry.c
     )
 ELSE()
     LIST(APPEND FREEGLUT_SRCS
