--- caca/driver/win32.c	2021-10-28 21:35:44.508000000 +0100
+++ caca/driver/win32.c	2021-10-28 21:35:47.115000000 +0100
@@ -23,7 +23,7 @@
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) && !defined(__clang__)
 /* This is missing from the MinGW headers. */
 #   if (_WIN32_WINNT >= 0x0500)
 BOOL WINAPI GetCurrentConsoleFont(HANDLE hConsoleOutput, BOOL bMaximumWindow,
@@ -33,7 +33,6 @@
 #ifndef MOUSE_HWHEELED
 #   define MOUSE_HWHEELED 0x0008
 #endif
-
 #include <stdlib.h>
 #include <stdio.h>
--- caca/caca_internals.h       2024-04-07 23:12:18.362347386 +0100
+++ caca/caca_internals.h       2024-04-07 23:11:53.625438477 +0100
@@ -268,7 +268,7 @@
 extern void _caca_set_term_title(char const *);
 
 /* Internal memory function */
-extern void *_caca_alloc2d(size_t width, size_t height, size_t elem_size);
+// extern void *_caca_alloc2d(size_t width, size_t height, size_t elem_size);
 
 /* Profiling functions */
 #if defined PROF
   GNU nano 7.2                                                                        caca/caca.h.patch                                                                                 
--- caca/caca.h 2024-04-07 23:15:21.881007088 +0100
+++ caca/caca.h 2024-04-07 23:15:23.513065740 +0100
@@ -47,6 +47,8 @@
 {
 #endif
 
+extern void *_caca_alloc2d(size_t width, size_t height, size_t elem_size);
+
 /** \e libcaca canvas */
 typedef struct caca_canvas caca_canvas_t;
 /** dither structure */
 