--- libde265/CMakeLists.txt	2022-02-03 01:20:49.520451800 +0000
+++ libde265/CMakeLists.txt	2022-02-03 01:15:11.370451800 +0000
@@ -86,6 +86,7 @@
 
 add_definitions(-DLIBDE265_EXPORTS)
 
+add_compile_options(-Wno-unused-variable -Wno-unused-but-set-variable -Wno-unused-function -Wno-uninitialized)
 add_subdirectory (encoder)
 
 if(NOT DISABLE_SSE)
@@ -110,18 +111,27 @@
 add_library(${PROJECT_NAME} ${libde265_sources} ${ENCODER_OBJECTS} ${X86_OBJECTS})
 target_link_libraries(${PROJECT_NAME} PRIVATE Threads::Threads)
 
+set(libde265_targets ${PROJECT_NAME})
+if(BUILD_SHARED_LIBS AND BUILD_STATIC_LIBS)
+  add_library(${PROJECT_NAME}-static STATIC ${libde265_sources} ${ENCODER_OBJECTS} ${X86_OBJECTS})
+  target_compile_definitions(${PROJECT_NAME}-static PRIVATE LIBDE265_STATIC_BUILD)
+  target_link_libraries(${PROJECT_NAME}-static PRIVATE Threads::Threads)
+  set_target_properties(${PROJECT_NAME}-static PROPERTIES OUTPUT_NAME ${PROJECT_NAME})
+  list(APPEND libde265_targets ${PROJECT_NAME}-static)
+endif()
 write_basic_package_version_file(${PROJECT_NAME}ConfigVersion.cmake COMPATIBILITY ExactVersion)
 
-install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config
+install(TARGETS ${libde265_targets} EXPORT ${PROJECT_NAME}Config
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
 install(FILES ${libde265_headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/de265-version.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+install(FILES ${CMAKE_SOURCE_DIR}/libde265/de265-version.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
 
 install(EXPORT ${PROJECT_NAME}Config DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake DESTINATION
-    "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
\ No newline at end of file
+    "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+install(FILES ${CMAKE_SOURCE_DIR}/libde265.pc DESTINATION lib/pkgconfig)

--- CMakeLists.txt	2022-02-03 01:20:09.220451800 +0000
+++ CMakeLists.txt	2022-02-03 01:22:43.040451800 +0000
@@ -46,7 +46,15 @@
   add_definitions(-DHAVE_POSIX_MEMALIGN)
 endif()
 
-configure_file (libde265/de265-version.h.in libde265/de265-version.h)
+configure_file(libde265/de265-version.h.in ${CMAKE_SOURCE_DIR}/libde265/de265-version.h)
+
+set(VERSION ${PROJECT_VERSION})
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix "\${prefix}")
+set(libdir "\${prefix}/lib")
+set(LIBS "-lpthread -lm")
+set(includedir "\${prefix}/include")
+configure_file(libde265.pc.in ${CMAKE_SOURCE_DIR}/libde265.pc @ONLY)
 
 if(CMAKE_COMPILER_IS_GNUCXX OR ${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
   add_definitions(-Wall)
@@ -55,9 +63,6 @@
 option(DISABLE_SSE "Disable SSE optimizations" OFF)
 
 option(BUILD_SHARED_LIBS "Build shared library" ON)
-if(NOT BUILD_SHARED_LIBS)
-  add_definitions(-DLIBDE265_STATIC_BUILD)
-endif()
 
 include_directories ("${PROJECT_SOURCE_DIR}")
 include_directories ("${PROJECT_BINARY_DIR}")
