--- CMakeLists.old	2021-11-02 10:11:28.156000000 +0000
+++ CMakeLists.txt	2021-11-02 10:12:41.268513000 +0000
@@ -180,7 +180,7 @@
 SET(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libraries.")
 
 OPTION(BUILD_STATIC_LIBS "Build static libraries." ON)
-
+OPTION(INSTALL_DOCS "Install documents and man pages" OFF)
 OPTION(PCRE2_BUILD_PCRE2_8 "Build 8 bit PCRE2 library" ON)
 
 OPTION(PCRE2_BUILD_PCRE2_16 "Build 16 bit PCRE2 library" OFF)
@@ -1027,10 +1027,11 @@
         SET(man3_new ${man3} ${man})
 ENDFOREACH(man ${man3})
 SET(man3 ${man3_new})
-
+IF(INSTALL_DOCS)
 INSTALL(FILES ${man1} DESTINATION man/man1)
 INSTALL(FILES ${man3} DESTINATION man/man3)
 INSTALL(FILES ${html} DESTINATION share/doc/pcre2/html)
+ENDIF(INSTALL_DOCS)
 
 IF(MSVC AND INSTALL_MSVC_PDB)
     INSTALL(FILES ${PROJECT_BINARY_DIR}/pcre2.pdb
