--- source/common/threadpool.c	2024-02-24 18:48:42.451907603 +0000
+++ source/common/threadpool.c	2024-02-24 18:49:28.358941470 +0000
@@ -98,7 +98,7 @@
 static INLINE
 int  xavs2_thread_set_cpu(int idx_core)
 {
-#if HAVE_POSIXTHREAD && (SYS_WINDOWS || SYS_LINUX) && !__MINGW32__
+#if HAVE_POSIXTHREAD && (SYS_WINDOWS || SYS_LINUX) && !(__MINGW32__ || __ANDROID__)
     cpu_set_t mask;
     CPU_ZERO(&mask);
 
@@ -118,7 +118,7 @@
 static INLINE
 int  xavs2_thread_is_on_cpu(int idx_core)
 {
-#if HAVE_POSIXTHREAD && (SYS_WINDOWS || SYS_LINUX) && !__MINGW32__
+#if HAVE_POSIXTHREAD && (SYS_WINDOWS || SYS_LINUX) && !(__MINGW32__ || __ANDROID__)
     cpu_set_t get;
 
     CPU_ZERO(&get);
--- source/encoder/encoder.c	2024-02-21 20:07:31.041383513 +0000
+++ source/encoder/encoder.c	2024-02-21 20:07:49.580605127 +0000
@@ -2038,7 +2038,7 @@
 
     /* start AEC frame coding */
     if (h->h_top->threadpool_aec != NULL && !h->param->enable_alf) {
-        xavs2_threadpool_run(h->h_top->threadpool_aec, encoder_aec_encode_one_frame, h, 0);
+        xavs2_threadpool_run(h->h_top->threadpool_aec, (xavs2_tfunc_t)encoder_aec_encode_one_frame, h, 0);
     }
 
     /* (3) encode all LCU rows in current frame ---------------------------
@@ -2147,7 +2147,7 @@
 #endif
 
         if (h->h_top->threadpool_aec != NULL) {
-            xavs2_threadpool_run(h->h_top->threadpool_aec, encoder_aec_encode_one_frame, h, 0);
+            xavs2_threadpool_run(h->h_top->threadpool_aec, (xavs2_tfunc_t)encoder_aec_encode_one_frame, h, 0);
         }
     }
 