--- libavcodec/allcodecs.c	2024-04-19 17:40:40.221860549 +0100
+++ libavcodec/allcodecs.c	2024-04-19 17:43:42.904366710 +0100
@@ -799,6 +799,8 @@
 extern const FFCodec ff_libvpx_vp8_decoder;
 extern FFCodec ff_libvpx_vp9_encoder;
 extern const FFCodec ff_libvpx_vp9_decoder;
+extern const FFCodec ff_libvvdec_decoder;
+extern const FFCodec ff_libvvenc_encoder;
 /* preferred over libwebp */
 extern const FFCodec ff_libwebp_anim_encoder;
 extern const FFCodec ff_libwebp_encoder;
