--- libavformat/movenc.c	2024-04-19 18:12:41.652734487 +0100
+++ libavformat/movenc.c	2024-04-19 18:13:10.943019805 +0100
@@ -6387,8 +6387,9 @@
             } else if(par->codec_id == AV_CODEC_ID_HEVC && par->extradata_size > 21) {
                 int nal_size_length = (par->extradata[21] & 0x3) + 1;
                 ret = ff_mov_cenc_avc_write_nal_units(s, &trk->cenc, nal_size_length, pb, pkt->data, size);
-            } else if(par->codec_id == AV_CODEC_ID_VVC) {
-                ret = AVERROR_PATCHWELCOME;
+            } else if(par->codec_id == AV_CODEC_ID_VVC && par->extradata_size > 21) {
+                int nal_size_length = (par->extradata[21] & 0x3) + 1;
+                ret = ff_mov_cenc_avc_write_nal_units(s, &trk->cenc, nal_size_length, pb, pkt->data, size);
             } else {
                 ret = ff_mov_cenc_write_packet(&trk->cenc, pb, pkt->data, size);
             }
